diff --git a/components/interactive/SignIn.astro b/components/interactive/SignIn.astro
index e4e2ab869f8106c9d86cbfeb380d3621c47cf391..d1d811ec727c616e33c6645dd925503d76ea52e3 100644
--- a/components/interactive/SignIn.astro
+++ b/components/interactive/SignIn.astro
@@ -3,6 +3,15 @@ import type { SignInProps } from "@clerk/types";
 type Props = SignInProps
 
 import InternalUIComponentRenderer from './InternalUIComponentRenderer.astro'
+
+const isNetlify = import.meta.env.IS_NETLIFY;
+const isNetlify2 = import.meta.env.NETLIFY;
 ---
 
+<div>IS NETLIFY CLIENT {import.meta.env.IS_NETLIFY}</div>
+<div>IS NETLIFY SERVER {isNetlify}</div>
+<br />
+<div>NETLIFY CLIENT {import.meta.env.NETLIFY}</div>
+<div>NETLIFY SERVER {isNetlify2}</div>
+<br />
 <InternalUIComponentRenderer {...Astro.props} component="sign-in" />
diff --git a/dist/index.js b/dist/index.js
index ea65fa6e50c26e32f0def7adfb5889af9d921d00..3b031662e75b6b9e8c201c2bf3a7a1df684a33c8 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -60,6 +60,7 @@ function createIntegration() {
           if (typeof clerkJSVariant !== "undefined" && clerkJSVariant !== "headless" && clerkJSVariant !== "") {
             logger.error('Invalid value for clerkJSVariant. Acceptable values are `"headless"`, `""`, and `undefined`');
           }
+          
           const internalParams = {
             ...params,
             sdkMetadata: {
@@ -83,7 +84,8 @@ function createIntegration() {
                 ...buildEnvVarFromOption(domain, "PUBLIC_CLERK_DOMAIN"),
                 ...buildEnvVarFromOption(clerkJSUrl, "PUBLIC_CLERK_JS_URL"),
                 ...buildEnvVarFromOption(clerkJSVariant, "PUBLIC_CLERK_JS_VARIANT"),
-                ...buildEnvVarFromOption(clerkJSVersion, "PUBLIC_CLERK_JS_VERSION")
+                ...buildEnvVarFromOption(clerkJSVersion, "PUBLIC_CLERK_JS_VERSION"),
+                "import.meta.env.IS_NETLIFY": JSON.stringify(process.env.NETLIFY !== undefined)
               },
               ssr: {
                 external: ["node:async_hooks"]
@@ -112,6 +114,17 @@ function createIntegration() {
             ${command === "dev" ? `console.log("${name}","Initialize Clerk: page")` : ""}
             import { runInjectionScript, swapDocument } from "${buildImportPath}";
 
+            // Remove param and update URL
+            function removeClerkQueryParam(param) {
+              const url = new URL(window.location.href);
+              if (url.searchParams.has(param)) {
+                url.searchParams.delete(param);
+                window.history.replaceState(window.history.state, '', url);
+              }
+            }
+
+            removeClerkQueryParam('__netlify_clerk_cache_bust');
+
             // Taken from https://github.com/withastro/astro/blob/e10b03e88c22592fbb42d7245b65c4f486ab736d/packages/astro/src/transitions/router.ts#L39.
             // Importing it directly from astro:transitions/client breaks custom client-side routing
             // even when View Transitions is disabled.
diff --git a/dist/server/index.js b/dist/server/index.js
index f65760365c60f46a00bb0949ff8fbaf0e14536db..109c30204e08af3b2b99e73f567a97c55be43203 100644
--- a/dist/server/index.js
+++ b/dist/server/index.js
@@ -3,7 +3,7 @@ import { verifyToken, createClerkClient as createClerkClient2 } from "@clerk/bac
 
 // src/server/clerk-middleware.ts
 import { AuthStatus as AuthStatus2, constants as constants3, createClerkRequest, createRedirect } from "@clerk/backend/internal";
-import { isDevelopmentFromSecretKey as isDevelopmentFromSecretKey2 } from "@clerk/shared/keys";
+import { isDevelopmentFromPublishableKey, isDevelopmentFromSecretKey as isDevelopmentFromSecretKey2 } from "@clerk/shared/keys";
 import { isHttpOrHttps } from "@clerk/shared/proxy";
 import { eventMethodCalled } from "@clerk/shared/telemetry";
 import { handleValueOrFn } from "@clerk/shared/utils";
@@ -177,6 +177,17 @@ var serverRedirectWithAuth = (context, clerkRequest, res, opts) => {
   return res;
 };
 
+function handleNetlifyCacheInDevInstance(locationHeader, requestState) {
+  if (import.meta.env.IS_NETLIFY && isDevelopmentFromPublishableKey(requestState.publishableKey)) {
+    const hasHandshakeQueryParam = locationHeader.includes('__clerk_handshake');
+    if (!hasHandshakeQueryParam) {
+      const url = new URL(locationHeader);
+      url.searchParams.append('__netlify_clerk_cache_bust', Date.now().toString());
+      requestState.headers.set('Location', url.toString());
+    }
+  }
+}
+
 // src/server/clerk-middleware.ts
 var CONTROL_FLOW_ERROR = {
   REDIRECT_TO_SIGN_IN: "CLERK_PROTECT_REDIRECT_TO_SIGN_IN"
@@ -199,8 +210,11 @@ var clerkMiddleware = (...args) => {
       clerkRequest,
       createAuthenticateRequestOptions(clerkRequest, options, context)
     );
+
     const locationHeader = requestState.headers.get(constants3.Headers.Location);
     if (locationHeader) {
+      handleNetlifyCacheInDevInstance(locationHeader, requestState);
+
       const res = new Response(null, { status: 307, headers: requestState.headers });
       return decorateResponseWithObservabilityHeaders(res, requestState);
     } else if (requestState.status === AuthStatus2.Handshake) {
